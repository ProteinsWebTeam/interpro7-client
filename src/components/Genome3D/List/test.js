// @flow
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import { List } from '.';

const renderer = new ShallowRenderer();

describe('<List />', () => {
  test('should render', () => {
    renderer.render(
      <List
        data={{
          loading: false,
          payload: {
            data: [
              {
                accession: 'E0W2I7',
                length: 1010,
                locations: [{ fragments: [{ start: 87, end: 179 }] }],
                metadata: {
                  anno_id: 581570,
                  confidence: '99',

                  resource: 'Gene3D',
                  type: 'PREDICTED_DOMAIN',
                },
                tooltipContent:
                  "Predicted domain from Genome3D resource 'Gene3D' based on UniProtKB E0W2I7/87-179 matching classification 1nl1A00",
              },
            ],
            interpro: {
              ipr_id: 'IPR000001',
              release_date: '2019-03-25T00:00:00',
              release_name: 'InterPro v73',
            },
            message:
              'Showing 1 to 20 of 1649 Genome3D annotations for InterPro entry IPR000001',
            pager: {
              current_page: 1,
              entries_per_page: 20,
              entry_from: 1,
              entry_to: 20,
              total_entries: 1649,
              total_pages: 83,
            },
          },
        }}
        customLocation={{
          description: {
            entry: {
              accession: 'IPR000001',
              db: 'InterPro',
              detail: 'genome3d',
            },
            main: {
              ket: 'entry',
            },
          },
        }}
        dataResource={{
          loading: false,
          payload: [],
        }}
      />,
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
