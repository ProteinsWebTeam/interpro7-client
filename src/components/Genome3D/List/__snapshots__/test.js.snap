// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> should render 1`] = `
<div
  className="row"
>
  <div
    className="columns"
  >
    <Memo(Loadable)
      data={
        Array [
          Object {
            "accession": "E0W2I7",
            "length": 1010,
            "locations": Array [
              Object {
                "fragments": Array [
                  Object {
                    "end": 179,
                    "start": 87,
                  },
                ],
              },
            ],
            "metadata": Object {
              "anno_id": 581570,
              "confidence": "99",
              "resource": "Gene3D",
              "type": "PREDICTED_DOMAIN",
            },
            "tooltipContent": "Predicted domain from Genome3D resource 'Gene3D' based on UniProtKB E0W2I7/87-179 matching classification 1nl1A00",
          },
        ]
      }
      processData={[Function]}
    />
    <Table
      actualSize={1649}
      dataTable={
        Array [
          Object {
            "accession": "E0W2I7",
            "anno_id": 581570,
            "confidence": "99",
            "id": 581570,
            "length": 1010,
            "locations": Array [
              Object {
                "fragments": Array [
                  Object {
                    "end": 179,
                    "start": 87,
                  },
                ],
              },
            ],
            "resource": "Gene3D",
            "tooltipContent": "Predicted domain from Genome3D resource 'Gene3D' based on UniProtKB E0W2I7/87-179 matching classification 1nl1A00",
            "type": "PREDICTED_DOMAIN",
          },
        ]
      }
      loading={false}
      notFound={false}
      rowKey="id"
    >
      <Memo(Connect(Exporter))>
        <div
          className="menu-grid"
        >
          <label
            htmlFor="json"
          >
            JSON
          </label>
          <Memo(Connect(FileExporter))
            count={1649}
            fileType="json"
            name="genome3d.IPR000001.json"
          />
          <label
            htmlFor="tsv"
          >
            TSV
          </label>
          <Memo(Connect(FileExporter))
            count={1649}
            fileType="tsv"
            name="genome3d.IPR000001.tsv"
          />
        </div>
      </Memo(Connect(Exporter))>
      <Column
        customiseSearch={
          Object {
            "message": "Accepts only UniProt accessions",
            "validation": /\\^\\(\\[OPQ\\]\\[0-9\\]\\[A-Z0-9\\]\\{3\\}\\[0-9\\]\\|\\[A-NR-Z\\]\\[0-9\\]\\(\\[A-Z\\]\\[A-Z0-9\\]\\{2\\}\\[0-9\\]\\)\\{1,2\\}\\)\\$/i,
          }
        }
        dataKey="accession"
        isSearchable={true}
        renderer={[Function]}
      >
        Protein
      </Column>
      <Column
        dataKey="evidences"
        isSearchable={true}
        options={Array []}
        renderer={[Function]}
        showOptions={true}
      >
        Evidence
      </Column>
      <Column
        customiseSearch={
          Object {
            "placeholder": ">=",
            "type": "number",
          }
        }
        dataKey="confidence"
        isSearchable={true}
        renderer={[Function]}
      />
      <Column
        dataKey="locations"
        renderer={[Function]}
      >
        Matches
      </Column>
    </Table>
  </div>
</div>
`;
